internal/app/networkd/pkg/nic/nic.go

```diff
@@ -404,7 +412,10 @@ func (n *NetworkInterface) configureInterface(method address.Addressing, link *n
 
 		err = n.rtnlConn.RouteAdd(method.Link(), *r.Dest, gw, rtnl.WithRouteSrc(src), rtnl.WithRouteAttrs(attr))
 		if err != nil {
-			return err
+			// ignore "EEXIST" errors for routes which are already present
+			if opErr, ok := err.(*netlink.OpError); !ok || !os.IsExist(opErr.Err) {
```

Comment:
nit: maybe split these two cases (!ok and !os.IsExist) to make them easier to human parse... moreover, I think I would have differing error messaging for each.

Reply:
we only ignore one specific error here, and any other error is going to be returned to the caller